sycl/include/CL/sycl/ONEAPI/group_algorithm.hpp

```diff
@@ -548,26 +552,23 @@ EnableIfIsNonNativeOp<T, BinaryOperation> reduce(Group g, V x, T init,
 }
 
 template <typename Group, typename Ptr, class BinaryOperation>
-EnableIfIsPointer<Ptr, typename Ptr::element_type>
+detail::enable_if_t<
+    (detail::is_generic_group<Group>::value && detail::is_pointer<Ptr>::value &&
+     detail::is_arithmetic<typename detail::remove_pointer<Ptr>::type>::value),
+    typename detail::remove_pointer<Ptr>::type>
 reduce(Group g, Ptr first, Ptr last, BinaryOperation binary_op) {
-  static_assert(sycl::detail::is_generic_group<Group>::value,
-                "Group algorithms only support the sycl::group and "
-                "ONEAPI::sub_group class.");
+  using T = typename detail::remove_pointer<Ptr>::type;
   // FIXME: Do not special-case for half precision
   static_assert(
-      std::is_same<decltype(binary_op(*first, *first)),
-                   typename Ptr::element_type>::value ||
-          (std::is_same<typename Ptr::element_type, half>::value &&
+      std::is_same<decltype(binary_op(*first, *first)), T>::value ||
```

Comment:
this change does not look connected to PR description. Could you please add the motivation there?

Reply:
Yes, I'll add a note there.  While updating the traits to use `is_pointer` I realized that the implementation as written would only support `multi_ptr` because of the reliance on `Ptr::element_type`.

Reply:
no extra comment is needed as you provide description in PR